alphalens/utils.py

```diff
@@ -853,12 +892,26 @@ def std_conversion(period_std, base_period):
     return period_std / np.sqrt(conversion_factor)
 
 
-def get_forward_returns_columns(columns):
+def get_forward_returns_columns(columns, require_exact_day_multiple=False):
     """
     Utility that detects and returns the columns that are forward returns
     """
-    pattern = re.compile(r"^(\d+([Dhms]|ms|us|ns))+$", re.IGNORECASE)
-    valid_columns = [(pattern.match(col) is not None) for col in columns]
+
+    # If exact day multiples are required in the forward return periods,
+    # drop all other columns (e.g. drop 3D12h).
+    if require_exact_day_multiple:
+        pattern = re.compile(r"^(\d+([D]))+$", re.IGNORECASE)
+        valid_columns = [(pattern.match(col) is not None) for col in columns]
+        
+        if sum(valid_columns) < len(valid_columns):
+            warnings.warn(
+                "Skipping return periods that aren't exact multiples" \
+                + " of days."
```

Comment:
I wonder if this has to be a warning (?). The users will be probably puzzled. Not a big deal anyway, just my 2 cents

Reply:
My thought was that if the user provides return periods that aren't expecting exact multiples of a day, they might be surprised if they don't see them in the tearsheets. I can remove it if we don't expect this to be a surprise.